addons/website_forum/tests/test_controller.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+from odoo.tests.common import HttpCase, new_test_user
+
+
+class TestController(HttpCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        portal_user = new_test_user(cls.env, login='portal_user', groups='base.group_portal')
+        cls.portal = portal_user.login
+        admin_user = new_test_user(cls.env, login='admin_user', groups='base.group_user,base.group_system')
+        cls.admin = admin_user.login
+
+        cls.headers = {"Content-Type": "application/json"}
+
+        cls.error = "odoo.exceptions.AccessError"
+
+    def _build_payload(self, params=None):
+        """
+        Helper to properly build jsonrpc payload
+        """
+        return {
+            "jsonrpc": "2.0",
+            "method": "call",
+            "id": 0,
+            "params": params or {},
+        }
+
+    def test_portal_fetch_attachment(self):
```

Comment:
Nitpicking/can ignore: We usually try to add number to the methods, `test_01_xx` etc.